web-client/src/presenter/actions/setWaitingForResponseAction.js

```diff
@@ -9,4 +9,5 @@ import { state } from 'cerebral';
 export const setWaitingForResponseAction = ({ store }) => {
   store.increment(state.progressIndicator.waitingForResponseRequests);
   store.set(state.progressIndicator.waitingForResponse, true);
+  store.unset(state.progressIndicator.waitText);
```

Comment:
I'm curious why this line was added? It seems like this action is called immediately before we set the wait text

Reply:
so the reason we added this here is because when you do a XHR request, the spinner will show up and display the waitText if it was set from the action previously in the sequence.  So if you navigate to a new page and do a different XHR request, the text will stick around on the next time we display a spinner.  We only wanted to add this spinner text to this specific user flow and not to every spinner.

Reply:
So is that why it's not unset in the `unsetWaitingForResponseAction`?

Reply:
Would this be a good time to use some inline comments to explain _why_ we unset this?  I'll defer to others who understand the implications more than I.

Reply:
we refactored this to be placed in the unsetWaitingForResponseAction since we think it makes more sense there during cleanup.

Reply:
I think I'd prefer a named test if possible